### PR TITLE
feat(NX-3132): Add purchase on inquiry button

### PR DESCRIPTION
**Pull request by lean8086**
The type of this PR is: **Feature**

This PR solves [NX-3132]

### Description

- Create `PurchaseOnInquiryButton` component
- Clean up `MakeOfferOnInquiryButton` to be only a button: move guarantee message to `ConversationCTA`
- Make `ConfirmArtworkButton` able to create both types of orders (buy and offer)

image
image

cc @artsy/negotiate-devs 

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[NX-3132]: https://artsyproduct.atlassian.net/browse/NX-3132?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Apps/Conversation/Components/ConfirmArtworkButton.tsx [hunk 1]
```diff
@@ -2,65 +2,62 @@ import { useState } from "react"
...
 import createLogger from "v2/Utils/logger"
 import { MakeInquiryOffer } from "../Mutation/MakeInquiryOfferMutation"
+import { MakeInquiryOrder } from "../Mutation/MakeInquiryOrderMutation"
 
 import { ConfirmArtworkButton_artwork } from "v2/__generated__/ConfirmArtworkButton_artwork.graphql"
 
 const logger = createLogger("Conversation/Components/ConfirmArtworkButton.tsx")
 
-export interface ConfirmArtworkButtonProps {
+export interface ConfirmArtworkButtonProps extends ButtonProps {
```
Comment (lean8086) [inline src/v2/Apps/Conversation/Components/ConfirmArtworkButton.tsx:16]: I needed to add `variant` as well, so it was a good opportunity to replicate the job made on Eigen.

File: src/v2/Apps/Conversation/Components/ConfirmArtworkButton.tsx [hunk 2]
```diff
@@ -2,65 +2,62 @@ import { useState } from "react"
...
+  trackingEvent,
+  createsOfferOrder = true,
+  disabled,
+  variant,
+  children,
+}) => {
+  const tracking = useTracking()
+  const [isCommittingMutation, setIsCommittingMutation] = useState(false)
 
+  const commitMutation = createsOfferOrder ? MakeInquiryOffer : MakeInquiryOrder
```
Comment (lean8086) [inline src/v2/Apps/Conversation/Components/ConfirmArtworkButton.tsx:39]: The important change on this file is that now it's able to create both buy and offer orders.

File: src/v2/Apps/Conversation/Components/ConversationCTA.tsx [hunk 3]
```diff
@@ -27,16 +51,46 @@ export const ConversationCTA: React.FC<ConversationCTAProps> = ({
...
+            <Flex flexShrink={1}>
+              <Text color="black60" variant="xs" mb={1}>
+                Always complete purchases with our secure checkout in order to
+                be covered by{" "}
+                <RouterLink to="/buyer-guarantee" target="_blank">
+                  The Artsy Guarantee
+                </RouterLink>
+                .
+              </Text>
+            </Flex>
```
Comment (lean8086) [inline src/v2/Apps/Conversation/Components/ConversationCTA.tsx:76]: The Guarantee message was moved here so the only responsibility of `MakeOfferOnInquiryButton` is to be a button.

File: src/v2/Apps/Conversation/Components/PurchaseOnInquiryButton.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+export interface PurchaseOnInquiryButtonProps {
+  conversation: PurchaseOnInquiryButton_conversation
+}
+
+const trackTappedPurchase = (id: string): TappedBuyNow => ({
+  action: ActionType.tappedBuyNow,
+  context_owner_type: OwnerType.conversation,
+  // @ts-ignore
+  impulse_conversation_id: id,
```
[the remark this answers is not in the document]
Comment (lean8086) [reply src/v2/Apps/Conversation/Components/PurchaseOnInquiryButton.tsx:16]: nvm, this is under a feature flag anyways

File: src/v2/Apps/Conversation/Components/ConfirmArtworkModal.tsx [hunk 5]
```diff
@@ -59,10 +61,12 @@ export const ConfirmArtworkModal: React.FC<ConfirmArtworkModalProps> = ({
...
-            trackingEvent={{
-              context_module: OwnerType.conversationMakeOfferConfirmArtwork,
-              context_owner_type: OwnerType.conversation,
-            }}
+            onClick={() =>
+              tracking.trackEvent({
+                context_module: OwnerType.conversationMakeOfferConfirmArtwork,
+                context_owner_type: OwnerType.conversation,
+              })
+            }
```
Comment (jo-rs) [inline src/v2/Apps/Conversation/Components/ConfirmArtworkModal.tsx:69]: Why was this moved from the child component if we don't have to get any data from the parent? 🤔

File: src/v2/Apps/Conversation/Components/ConversationCTA.tsx [hunk 6]
```diff
@@ -67,6 +121,8 @@ export const ConversationCTAFragmentContainer = createFragmentContainer(
             ... on Artwork {
               __typename
               isOfferableFromInquiry
+              is_acquireable: isAcquireable
```
Comment (jo-rs) [inline src/v2/Apps/Conversation/Components/ConversationCTA.tsx:124]: Is there a reason for this to be aliased? 🤔

File: src/v2/Apps/Conversation/Components/MakeOfferOnInquiryButton.tsx [hunk 7]
```diff
@@ -101,6 +77,7 @@ export const MakeOfferOnInquiryButtonFragmentContainer = createFragmentContainer
           liveArtwork {
             ... on Artwork {
               __typename
+              is_acquireable: isAcquireable
```
Comment (jo-rs) [inline src/v2/Apps/Conversation/Components/MakeOfferOnInquiryButton.tsx:80]: Same question about this being aliased.

**Review by araujobarret — APPROVED** [review]
Everything looks good, great tests though, testing if mutations are called is clean 🎸

File: src/v2/Apps/Conversation/Components/ConfirmArtworkModal.tsx (hunk 5 above)
Comment (araujobarret) [reply src/v2/Apps/Conversation/Components/ConfirmArtworkModal.tsx:69]: I guess because of the way we use it, we use it from `MakeOfferOnInquiryButton` which dispatches a different event than this modal here

Comment (lean8086) [reply src/v2/Apps/Conversation/Components/ConfirmArtworkModal.tsx:69]: The child component `ConfirmArtworkButton` needs to track different events depending on the context (confirm artwork, click purchase, click make offer).
In this case, the ConfirmArtworkModal is tracking the "confirm artwork" event. So, it's not because of the data it needs, but because of who triggers what. Thoughts?

File: src/v2/Apps/Conversation/Components/ConversationCTA.tsx (hunk 6 above)
Comment (lean8086) [reply src/v2/Apps/Conversation/Components/ConversationCTA.tsx:124]: Good question. We are using it like this in many components related to the artwork page and I was just following that. If this is not necessary I'll remove it!
